src/libPMacc/include/traits/GetUniqueTypeId.hpp

```diff
@@ -48,14 +48,21 @@ namespace detail
 template<typename T_Type>
 struct GetUniqueTypeId
 {
-    static uint8_t byte;
+    static uint64_t counter;
     static const uint64_t id;
 };
+
+template<>
+const uint64_t GetUniqueTypeId<uint8_t>::id = 0;
+template<>
+uint64_t GetUniqueTypeId<uint8_t>::counter = 0;
+
 template<typename T_Type>
-uint8_t GetUniqueTypeId<T_Type>::byte;
+const uint64_t GetUniqueTypeId<T_Type>::id = ++GetUniqueTypeId<uint8_t>::counter;
```

Comment:
Doesnt this result in all types having the same value for id equal 1?
Do I overlook something?